Doc/reference/simple_stmts.rst

```diff
@@ -806,6 +806,14 @@ Examples::
    from foo.bar import baz    # foo.bar.baz imported and bound as baz
    from foo import attr       # foo imported and foo.attr bound as attr
 
+Note: When a submodule is imported, all parent modules are imported and may
```

Comment:
I think this is too much detail. Above there is already a reference to https://docs.python.org/3/reference/import.html#searching, which explains this behavior. I would just change the comments in the above block to say "foo, foo.bar, and foo.bar.baz imported".